R-package/tests/testthat/test_basic.R

```diff
@@ -188,6 +188,67 @@ test_that("lightgbm() rejects negative or 0 value passed to nrounds", {
   }
 })
 
+test_that("lightgbm() accepts nrounds as either a top-level argument or parameter", {
+  nrounds <- 15L
+
+  set.seed(708L)
+  top_level_bst <- lightgbm(
+    data = train$data
+    , label = train$label
+    , nrounds = nrounds
+    , params = list(
+      objective = "regression"
+      , metric = "l2"
+      , num_leaves = 5L
+    )
+    , save_name = tempfile(fileext = ".model")
+  )
+
+  set.seed(708L)
+  param_bst <- lightgbm(
+    data = train$data
+    , label = train$label
+    , params = list(
+      objective = "regression"
+      , metric = "l2"
+      , num_leaves = 5L
+      , nrounds = nrounds
+    )
+    , save_name = tempfile(fileext = ".model")
+  )
+
+  set.seed(708L)
+  both_customized <- lightgbm(
+    data = train$data
+    , label = train$label
+    , nrounds = 20L
+    , params = list(
+      objective = "regression"
+      , metric = "l2"
+      , num_leaves = 5L
+      , nrounds = nrounds
+    )
+    , save_name = tempfile(fileext = ".model")
+  )
+  
+  top_level_l2 <- top_level_bst$eval_train()[[1L]][["value"]]
+  param_l2    <- param_bst$eval_train()[[1L]][["value"]]
+  both_l2      <- both_customized$eval_train()[[1L]][["value"]]
+
+  # check type just to be sure the subsetting didn't return a NULL
+  expect_true(is.numeric(top_level_l2))
+  expect_true(is.numeric(param_l2))
+  expect_true(is.numeric(both_l2))
+  
+  # check that model produces identical performance
+  expect_identical(top_level_l2, param_l2)
+  expect_identical(both_l2, param_l2)
+
+  expect_identical(param_bst$current_iter(), top_level_bst$current_iter())
```

Comment:
sorry for missing this in my last review! Could you also add an assertion here somewhere comparing one of these `$current_iter()` values to `nrounds` directly?

That way, bugs in `$current_iter()` won't be able to silently pass this test.

For example, if `$current_iter()` always returned `100`, all these assertions would pass, even though you used `nrounds = 20`.